exo-go/src/types/docker_config.go

```diff
@@ -4,7 +4,7 @@ package types
 // docker-compose.yml
 type DockerConfig struct {
 	Image         string            `yaml:",omitempty"`
-	Build         string            `yaml:",omitempty"`
+	Build         map[string]string `yaml:",omitempty"`
```

Comment:
It wasn't wrong, this `docker-compose` field could either be a string or a map, see [here](https://docs.docker.com/compose/compose-file/#context). Since  `exo-test` uses different `Dockerfile`s from `exo-run`. I need those fields to provide the right context for the commands in the `Dockerfile`s because I ran into some permission errors.

> This also feels like something that should be updated in its own PR

You're right but I thought it also made sense to have one PR for updating the `types` and config helpers functions so that we can handle running tests. I didn't expect the latter to be this much code. Sorry about that.